### PR TITLE
Automation for moving assets from bintray to github

### PR DESCRIPTION
**Pull request by ragsns**

### CODE REVIEW

File: before_deploy.sh [hunk 1]
```diff
@@ -2,13 +2,22 @@
...
 export BACKINGSERVICES_FILE_NAME=backingservices-${CHART_VERSION}.tgz
 cat descriptor-template.json | jq '.files[0].includePattern=env.PEGA_FILE_NAME' | jq '.files[0].uploadPattern=env.PEGA_FILE_NAME' | jq '.files[1].includePattern=env.ADDONS_FILE_NAME' | jq '.files[1].uploadPattern=env.ADDONS_FILE_NAME' | jq '.files[2].includePattern=env.BACKINGSERVICES_FILE_NAME' | jq '.files[2].uploadPattern=env.BACKINGSERVICES_FILE_NAME' > descriptor.json
-curl -o index.yaml https://dl.bintray.com/pegasystems/pega-helm-charts/index.yaml
+# Get the latest index.yaml from github.io
+curl -o index.yaml https://pegasystems.github.io/pega-helm-charts/index.yaml
+# Clone the versions from gh-pages to a temp directory - xyz
+# The versions will be re-installed in gh-pages
+git clone -b gh-pages https://github.com/${TRAVIS_REPO_SLUG} xyz
+cp xyz/* .
+rm -rf xyz
```
Comment (MadhuriArugula) [inline before_deploy.sh:17]: @ragsns Can you please explain why these steps are needed?

Comment (ragsns) [reply before_deploy.sh]: We reviewed this together and made some improvements.

File: before_deploy.sh [hunk 2]
```diff
@@ -7,8 +7,16 @@ export PEGA_FILE_NAME=pega-${CHART_VERSION}.tgz
 export ADDONS_FILE_NAME=addons-${CHART_VERSION}.tgz
 export BACKINGSERVICES_FILE_NAME=backingservices-${CHART_VERSION}.tgz
 cat descriptor-template.json | jq '.files[0].includePattern=env.PEGA_FILE_NAME' | jq '.files[0].uploadPattern=env.PEGA_FILE_NAME' | jq '.files[1].includePattern=env.ADDONS_FILE_NAME' | jq '.files[1].uploadPattern=env.ADDONS_FILE_NAME' | jq '.files[2].includePattern=env.BACKINGSERVICES_FILE_NAME' | jq '.files[2].uploadPattern=env.BACKINGSERVICES_FILE_NAME' > descriptor.json
-curl -o index.yaml https://dl.bintray.com/pegasystems/pega-helm-charts/index.yaml
+# Get the latest index.yaml from github.io
+curl -o index.yaml https://pegasystems.github.io/pega-helm-charts/index.yaml
+# Clone the versions from gh-pages to a temp directory - xyz
+# The versions will be re-installed in g temporary directory - temp_gh_pages
```
Comment (MadhuriArugula) [inline before_deploy.sh:13]: A slight documentation error. Can this be updated to - 
```
# Clone the versions from gh-pages to a temp directory - temp_gh_pages
# The versions will be re-installed into temporary directory - temp_gh_pages
```

**Review by kishorv10 — COMMENTED** [review]
@ragsns We are trying to understand the travis `pages` provider. How does it know about the files to be committed/uploaded to `gh-pages` branch. For example, with bintray there is [descriptor-template.json](https://github.com/pegasystems/pega-helm-charts/blob/master/descriptor-template.json). This kind of information is not mentioned. Is it going to commit just helm packages & index.yaml or entire master source code to `gh-pages` branch?

**Comment by ragsns** [comment]
@kishorv10 The gh-pages (which currently only consists of the artifacts for the helm charts i.e. what's available via github.io) is what's going to be the commit. So, the new .tgz s and the updated index.yaml. Without the git clone of the gh-pages branch only the new versions will be committed and the old versions will be removed and we don't want that (as I realized during my testing).

**Review by APegaDavis — APPROVED** [review]

**Review by MadhuriArugula — APPROVED** [review]

**Pull request merged**